lib/conn_parser.ex

```diff
@@ -1,62 +1,80 @@
 defmodule Xcribe.ConnParser do
+  @moduledoc ~S"""
+  Used to convert a `Plug.Conn` to a `Xcribe.Request`.
+
+  Each connection sent to documenting in your tests is given to `ConnParser`.
+  Is expected that connection has been passed through the app `Endpoint` as a
+  finished request. The parser will extract all needed info from `Conn` and uses
+  app `Router` for additional information about the request.
+
+  The atribute `description` must be given at `document` macro call with the
+  option `:as`:
+
+      test "test name", %{conn: conn} do
+        ...
+
+        Xcribe.Helpers.Document.document(conn, as: "description here")
+
+        ...
+      end
+
+  If no description is given the current test description will be used.
+  """
+
   alias Xcribe.{Config, Request}
 
-  def execute(conn, description \\ "sample request") do
-    route = identify_route(conn)
-    path = format_path(route.path, Map.keys(conn.path_params))
-    namespaces = fetch_namespaces()
+  @doc """
+  Parse the given `Plug.Conn` and transform it to a `Xcribe.Request`. A
+  description can be provided.
+  """
+  def execute(conn, description \\ "") do
+    conn
+    |> identify_route()
+    |> parse_conn(conn, description)
+  end
+
+  defp parse_conn({:error, _} = error, _conn, _description), do: error
+
+  defp parse_conn(route, conn, description) do
+    path = format_path(route.route, Map.keys(conn.path_params))
```

Comment:
What do you think?

```elixir
params = Map.keys(conn.path_params)
path = format_path(route.route, params)
```

Reply:
As I'll use params keys just for build the path I prefer not create a var just for it. So I moved the `Map.keys` to `format_path` function.